structure.md

```diff
@@ -0,0 +1,181 @@
+```txt
+com/example/ss2backend/
+    ├── aspects/                          # Cross-cutting concerns (e.g., logging)
+    ├── config/                           # Application configuration files
+    ├── controller/                       # Controllers for handling requests
+    │   └── StableMatchingController.java
+    ├── dto/                              # Data transfer objects (DTOs)
+    │   ├── StableMatchingRequestDTO.java
+    │   └── StableMatchingResponseDTO.java
+    ├── exception/                        # Custom exceptions for error handling
+    │   ├── InvalidPreferenceException.java
+    │   └── MatchingNotFoundException.java
+    ├── model/                            
+    │   ├── gt/                           # Game theory entities
+    │   └── smt/                          # Stable matching entities
+    │       ├── individual/              
+    │       │   ├── Individual.java
+    │       │   ├── IndividualList.java
+    │       │   └── PreferenceList.java
+    │       ├── match/   
+    │       │   ├── Matcher.java           # Interface or abstract class for matching algorithms, manage the actual matching process, including handling preferences and matches.
```

Comment:
Nghiên cứu về việc dùng chung một class Matches, không chia nhỏ các case, có thể xem xét sử dụng Map thay vì Vector nếu performance tốt hơn -> Ảnh hưởng: FE có thể sẽ phải sửa cách dịch kết quả từ BE do Map có thể sẽ không đảm bảo thứ tự.